File: hw/pazmino/u2/HW21Mongo/MongoDBC/src/main/java/utils/MongoDBManager.java
package utils;

import com.mongodb.MongoClientSettings;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoDatabase;
import com.mongodb.client.MongoCollection;
import org.bson.Document;

/**
 *
 * @author Brandon PAzmino
 */
public class MongoDBManager {
        private static final String CONNECTION_URI = "mongodb+srv://bpazmino:bpazmino@cluster0.ot5jx.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0"; // Reemplaza con tu URI
    private static final String DATABASE_NAME = "OOP";   // Nombre de tu base de datos

    private MongoClient mongoClient;
    private MongoDatabase database;

    public MongoDBManager() {
        try {
            // Inicializamos el cliente Mongo con el URI
            mongoClient = MongoClients.create(CONNECTION_URI);
            database = mongoClient.getDatabase(DATABASE_NAME);
        } catch (Exception e) {
            System.out.println("Error connecting to MongoDB: " + e.getMessage());
        }
    }

    public MongoCollection<Document> getCollection(String collectionName) {
        return database.getCollection(collectionName);
    }

    public void closeConnection() {
        if (mongoClient != null) {
            mongoClient.close();
        }
    }
    
}

File: EXAM/pazmino/u2/Q71-100/Students/utils/mongoDB_manager.py
from pymongo import MongoClient
from pymongo.errors import ConnectionError

class MongoDBManager:
    CONNECTION_URI = "mongodb+srv://bpazmino:bpazmino@cluster0.ot5jx.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0"  
    DATABASE_NAME = "OOP"  

    def __init__(self):
        try:
            self.mongo_client = MongoClient(self.CONNECTION_URI)
            self.database = self.mongo_client[self.DATABASE_NAME]
        except ConnectionError as e:
            print("Error connecting to MongoDB:", e)

    def get_collection(self, collection_name):
        return self.database[collection_name]

    def close_connection(self):
        if self.mongo_client:
            self.mongo_client.close()

# Ejemplo de uso
if __name__ == "__main__":
    mongo_manager = MongoDBManager()
    collection = mongo_manager.get_collection("tu_coleccion") 
 

    mongo_manager.close_connection()